File: SYP.Web/Modules/Setting/VendorType/VendorTypeColumns.cs
namespace SYP.Setting.Columns;

[ColumnsScript("Setting.VendorType")]
[BasedOnRow(typeof(VendorTypeRow), CheckNames = true)]
public class VendorTypeColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Title { get; set; }
    public string DiscountType { get; set; }
    public decimal DiscountValue { get; set; }
    public string Description { get; set; }
    public bool IsActive { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Setting/VendorTypeColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { VendorTypeRow } from "./VendorTypeRow";

export interface VendorTypeColumns {
    Id: Column<VendorTypeRow>;
    Title: Column<VendorTypeRow>;
    DiscountType: Column<VendorTypeRow>;
    DiscountValue: Column<VendorTypeRow>;
    Description: Column<VendorTypeRow>;
    IsActive: Column<VendorTypeRow>;
}

export class VendorTypeColumns extends ColumnsBase<VendorTypeRow> {
    static readonly columnsKey = 'Setting.VendorType';
    static readonly Fields = fieldsProxy<VendorTypeColumns>();
}